src/api.rs

```diff
@@ -106,43 +109,47 @@ impl NemoControl {
 mod test {
     use std::fs::read_to_string;
 
-    use super::*;
     use assert_fs::TempDir;
+    use serial_test::serial;
     use test_log::test;
 
+    use super::*;
+
     #[cfg_attr(miri, ignore)]
     #[test]
-    fn reason() {
+    #[serial]
+    fn reason2() {
```

Comment:
What is this test for? It seems you are testing the same code twice? These tests should make sure the API works as expected, not do full-scale integration tests (we do those elsewhere already). In particular, what there _should_ be tests for (actually not, because we really rather should get rid of the state object) is that, e.g., calling API methods in the wrong order returns _the expected errors_.